src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpFormattingOptions2.cs

```diff
@@ -125,43 +114,57 @@ private static Option2<bool> CreateNewLineForBracesOption(NewLineOption newLineO
         public static Option2<bool> SpacingAfterMethodDeclarationName { get; } = CreateOption(
             CSharpFormattingOptionGroups.Spacing, nameof(SpacingAfterMethodDeclarationName),
             CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.AfterMethodDeclarationName),
-            EditorConfigStorageLocation.ForBoolOption("csharp_space_between_method_declaration_name_and_open_parenthesis"),
+            new EditorConfigStorageLocation<bool>(CSharpEditorConfigSettingsValueHolder.SpacingAfterMethodDeclarationName.GetSettingName(),
+                                                  CSharpEditorConfigSettingsValueHolder.SpacingAfterMethodDeclarationName.GetValueFromEditorConfigString,
+                                                  CSharpEditorConfigSettingsValueHolder.SpacingAfterMethodDeclarationName.GetEditorConfigStringFromValue),
             new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpacingAfterMethodDeclarationName"));
 
         public static Option2<bool> SpaceWithinMethodDeclarationParenthesis { get; } = CreateOption(
             CSharpFormattingOptionGroups.Spacing, nameof(SpaceWithinMethodDeclarationParenthesis),
             CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.WithinMethodDeclarationParenthesis),
-            EditorConfigStorageLocation.ForBoolOption("csharp_space_between_method_declaration_parameter_list_parentheses"),
+            new EditorConfigStorageLocation<bool>(CSharpEditorConfigSettingsValueHolder.SpaceWithinMethodDeclarationParenthesis.GetSettingName(),
+                                                  CSharpEditorConfigSettingsValueHolder.SpaceWithinMethodDeclarationParenthesis.GetValueFromEditorConfigString,
+                                                  CSharpEditorConfigSettingsValueHolder.SpaceWithinMethodDeclarationParenthesis.GetEditorConfigStringFromValue),
             new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinMethodDeclarationParenthesis"));
 
         public static Option2<bool> SpaceBetweenEmptyMethodDeclarationParentheses { get; } = CreateOption(
             CSharpFormattingOptionGroups.Spacing, nameof(SpaceBetweenEmptyMethodDeclarationParentheses),
             CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.BetweenEmptyMethodDeclarationParentheses),
-            EditorConfigStorageLocation.ForBoolOption("csharp_space_between_method_declaration_empty_parameter_list_parentheses"),
+            new EditorConfigStorageLocation<bool>(CSharpEditorConfigSettingsValueHolder.SpaceBetweenEmptyMethodDeclarationParentheses.GetSettingName(),
```

Comment:
you should add an overload to `EditorConfigStorageLocation` ctor to just take in the EditorConfigData object - then internally call the right methods.  That way you just pass in `CSharpEditorConfigSettingsValueHolder.SpaceBetweenEmptyMethodDeclarationParentheses` once instead of multiple times.  This can help prevent bugs where you acidetnally pass two different settings